zulip/integrations/slack/slackdata2zulipdata.py

```diff
@@ -384,8 +386,67 @@ def check_user_mention(text):
     print('ls', os.listdir())
     print('pwd', os.getcwd())
 
-    # TODO
-    # attachments
+    # for attachment.json
+    attachment_json = {}
+    zerver_attachment = []
+
+    # upload folder : zulipdata/uploads/realm_id
+    upload_dir = output_dir + "/uploads/" + str(REALM_ID)
+    # create uploads directory to save all files in that directory
+    os.makedirs(upload_dir)
+
+    # To convert url in form https:\/\/files.slack.com\/files-pri\/..  to https://files.slack.com//files-pri//..
+    def parse_url(url):
+        url = url.replace("\/\/", "//")
+        url = url.replace("\/", "//")
+        return url
+
+    # Download from the slack url and save it to the folder zulipdata/uploads/realm_id/id/filename
+    def save_attachment(url, id, name):
+        url = parse_url(url)
+        response = requests.get(url, stream=True)
+        try:
+            os.makedirs(upload_dir + '/' + str(id))
+        except OSError:
+            print ('Directory already exists.')
+        with open(upload_dir + '/' + str(id) + '/' + name, 'wb') as output_file:
+            shutil.copyfileobj(response.raw, output_file)
+
+    # Get attachments by checking all messages
+    json_names = os.listdir(slack_dir + '/' + channel)
+    for json_name in json_names:
+        msgs = json.load(open(slack_dir + '/%s/%s' % (channel, json_name)))
+        attachment_id = 1
+        msg_id_count = 1
+        for msg in msgs:
```

Comment:
I think the implementation would be cleaner if the loop over ALL the messages happens only once, near L252 in the function `channelmessage2zerver_message_for_one_stream`. When the subtype is "file_share", do the `save_attachment` operation, and append to zerver_attachment.

Reply:
Right, I thought this too, but then even zerver_attachments would be a list of list of dict. ( list of dict of all attachments in a channel appended to the main list). The export data is just a list of dict of all the attachments. So what should I go ahead with?